### PR TITLE
form controller update

### PR DESCRIPTION
**Pull request by jnlaguerre23**
@levonlloyd please review form function in UserController.rb as well as form file in View/User

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -25,6 +25,12 @@ def new
   # GET /users/1/edit
   def edit
     @user = User.find(params[:id])
+
+  end
+
+  def form
```
Comment (levonlloyd) [inline app/controllers/users_controller.rb]: What is the purpose of this method?
@levonlloyd purpose is to find the current user in order for him to edit his profile

File: db/schema.rb [hunk 2]
```diff
@@ -11,7 +11,16 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20150317014350) do
+ActiveRecord::Schema.define(version: 20150413015135) do
+
+  create_table "forms", force: :cascade do |t|
```
Comment (levonlloyd) [inline db/schema.rb]: I think forms is a bad name for this.  This seems like data that is relevant to students.  Why not call it students?
@levonlloyd  Ok i can fix that instead of naming it "form"

**Comment by levonlloyd** [comment]
A couple of high level comments.
1) Usually, there is a controller that corresponds to each model object.  Here, you would add a StudentController, and Student model.  the student controller would have create and show methods and the form would post to StudentController#create which would create the student row in the db.
2) There is a student view that contains the actual form that is filled out by the user.
3) Please make sure tho delete the log/development.log and make sure that you don't add it back again.

basically, you want to add
StudentController with create and show methods to create and fetch Student objects
Student class in the model directory that has methods and validations of student objects
student view that has the html form that is to be filled out.

File: app/models/user.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@ class User < ActiveRecord::Base
   validates :email, presence: true, length: {maximum: 255},
             format: {with: VALID_EMAIL_REGEX},
             uniqueness: {case_sensitive: false}
+  validates :school, presence: true
```
Comment (levonlloyd) [inline app/models/user.rb]: this validation would belong on the student model class, not the user class.  THis class corresponds to rows in the user table.

File: app/models/user.rb [hunk 4]
```diff
@@ -22,6 +23,7 @@ def User.new_token
     SecureRandom.urlsafe_base64
   end
 
+
```
Comment (levonlloyd) [inline app/models/user.rb]: delete this extra line

File: db/schema.rb [hunk 5]
```diff
@@ -11,7 +11,16 @@
...
+    t.text     "school"
+    t.integer  "user_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```
Comment (levonlloyd) [inline db/schema.rb:20]: I think you also want a foreign key to the users table.  Each student is also a user correct?

**Comment by levonlloyd** [comment]
Also, please delete db/development.sqlite3 and add it to the .gitignore if it isn't already there.

**Comment by levonlloyd** [comment]
Also, unrelated to this change.  Do we store raw user password?  I hadn't seen that before, but if we do that is REALLY BAD.  Let me know if that is the case and we can create another change that changes that on Thursday.

**Comment by jnlaguerre23** [comment]
@levonlloyd i've made the changes and created a separate model, viewer and controller for student.

File: app/models/user.rb (hunk 3 above)
Comment (levonlloyd) [inline app/models/user.rb]: This still needs to be moved

**Comment by levonlloyd** [comment]
Please remove the tmp files and add the student model, view, and controller and the actual migraiton.

File: app/controllers/student_controller.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+class StudentController < ApplicationController
+
```
Comment (levonlloyd) [inline app/controllers/student_controller.rb:2]: please remove this extra space

File: db/migrate/20150415005734_add_school_to_student.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddSchoolToStudent < ActiveRecord::Migration
```
Comment (levonlloyd) [inline db/migrate/20150415005734_add_school_to_student.rb]: please combine this with the other migration into 1 migration

File: config/routes.rb [hunk 8]
```diff
@@ -2,10 +2,12 @@
     root                'static_pages#home'
     get    'help'    => 'static_pages#help'
     get    'about'   => 'static_pages#about'
+    get    'form'    => 'users#form'
```
Comment (levonlloyd) [inline config/routes.rb]: I think this needs to be 

``` ruby
get 'form' => 'student#create'
```

File: app/controllers/student_controller.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+
+  def new
+    @student=Student.new
+  end
+
+  def create
+    @student = User.find(params[:id])
```
Comment (levonlloyd) [inline app/controllers/student_controller.rb]: User.find finds a user, not a student.  You want to do something like

``` ruby
@student = Student.new
@user = User.find(session[:user_id]) # this gets the user's id from the session and finds the appropriate user object.
#fill in the student field from params
if @student.save
...
```
